sklearn/metrics/_plot/tests/test_plot_confusion_matrix.py

```diff
@@ -173,6 +173,9 @@ def test_confusion_matrix_display(pyplot, data, fitted_clf, y_pred, n_classes):
     image_data = disp.im_.get_array().data
     assert_allclose(image_data, cm)
 
+    disp.plot(colorbar=False)
+    assert disp.im_.colorbar is None
```

Comment:
I think that we should add a separate test to check the effect of the parameter.
Indeed, it is appearing in `disp.plot` and `plot_confusion_matrix` so we should check that the behaviour is correct.

```python
@pytest.mark.parametrize("colorbar, [True, False])
def test_plot_confusion_matrix_colorbar(colorbar):
    disp = plot_confusion_matrix(..., colorbar=colorbar)
    assert disp.im_.colorbar ...
    
     # attempt a plot with the opposit effect
     disp.plot(colorbar=not colorbar)
    assert disp.im_.colorbar ...
```

Reply:
This still needs to be addressed.